### PR TITLE
add run_constrained

### PR DESCRIPTION
**Pull request by dholth**
### Description

Is this spelled correctly?

TODO

- [x] https://github.com/conda-forge/conda-feedstock/pull/196
- [x] https://github.com/AnacondaRecipes/conda-feedstock/pull/21
- [ ] conda-forge hotfixes?

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -51,6 +51,7 @@ requirements:
     - conda-env >=2.6
     - conda-content-trust >=0.1.1
     - cytoolz >=0.8.1
+    - conda-libmamba-solver >=22.11.0
```
Comment (jezdez) [inline recipe/meta.yaml:54]:
```suggestion
    - conda-libmamba-solver >=22.12.0
```

**Review by jezdez — COMMENTED** [review]
LGTM!

**Review by jezdez — APPROVED** [review]

**Comment by mbargull** [comment]
This PR's changes are of course fine/good/complete and can be merged :).

----
But we should not ignore the CI errors from
https://github.com/conda/conda/actions/runs/3639899352/jobs/6143886123
and
https://github.com/conda/conda/actions/runs/3643394459/jobs/6154412333

----
For the first one on macOS we have
```
tests/test_activate.py::ShellWrapperIntegrationTests::test_bash_basic_integration FAILED 
tests/test_activate.py::ShellWrapperIntegrationTests::test_powershell_basic_integration FAILED
tests/test_activate.py::ShellWrapperIntegrationTests::test_zsh_basic_integration FAILED
```
With `test_bash_basic_integration` failing at
```
        shell.sendline('conda' + install + f'-yq hdf5={HDF5_VERSION}')
        shell.expect('Executing transaction: ...working... done.*\n', timeout=60)
```
with
```
2022-12-07T15:23:05.8429880Z conda install --dev -yq hdf5=1.12.1
2022-12-07T15:23:05.8430180Z Collecting package metadata (repodata.json): ...working... done
2022-12-07T15:23:05.8430460Z Solving environment: ...working... done
2022-12-07T15:23:05.8430600Z 
2022-12-07T15:23:05.8430690Z ## Package Plan ##
2022-12-07T15:23:05.8430820Z 
2022-12-07T15:23:05.8431010Z   environment location: /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/78c2 c5e3/envs/charizard
2022-12-07T15:23:05.8431210Z 
2022-12-07T15:23:05.8431310Z   added / updated specs:
2022-12-07T15:23:05.8431590Z     - hdf5=1.12.1
2022-12-07T15:23:05.8431710Z 
2022-12-07T15:23:05.8431710Z 
2022-12-07T15:23:05.8431830Z The following packages will be downloaded:
2022-12-07T15:23:05.8431990Z 
2022-12-07T15:23:05.8432290Z     package                    |            build
2022-12-07T15:23:05.8432670Z     ---------------------------|-----------------
2022-12-07T15:23:05.8433070Z     hdf5-1.12.1                |       h2b2ad87_2         5.6 MB
2022-12-07T15:23:05.8433590Z     libgfortran-5.0.0          |11_3_0_hecd8cb5_28         146 KB
2022-12-07T15:23:05.8433980Z     libgfortran5-11.3.0        |      h9dfd629_28         1.8 MB
2022-12-07T15:23:05.8434360Z     ------------------------------------------------------------
2022-12-07T15:23:05.8434650Z                                            Total:         7.5 MB
2022-12-07T15:23:05.8434790Z 
2022-12-07T15:23:05.8434900Z The following NEW packages will be INSTALLED:
2022-12-07T15:23:05.8435070Z 
2022-12-07T15:23:05.8435440Z   c-ares             pkgs/main/osx-64::c-ares-1.18.1-hca72f7f_0 
2022-12-07T15:23:05.8435950Z   ca-certificates    pkgs/main/osx-64::ca-certificates-2022.10.11-hecd8cb5_0 
2022-12-07T15:23:05.8436440Z   hdf5               pkgs/main/osx-64::hdf5-1.12.1-h2b2ad87_2 
2022-12-07T15:23:05.8436870Z   krb5               pkgs/main/osx-64::krb5-1.19.2-hcd88c3b_0 
2022-12-07T15:23:05.8437330Z   libcurl            pkgs/main/osx-64::libcurl-7.86.0-ha585b31_0 
2022-12-07T15:23:05.8437790Z   libcxx             pkgs/main/osx-64::libcxx-14.0.6-h9765a3e_0 
2022-12-07T15:23:05.8438250Z   libedit            pkgs/main/osx-64::libedit-3.1.20210910-hca72f7f_0 
2022-12-07T15:23:05.8438710Z   libev              pkgs/main/osx-64::libev-4.33-h9ed2024_1 
2022-12-07T15:23:05.8439640Z   libgfortran        pkgs/main/osx-64::libgfortran-5.0.0-11_3_0_hecd8cb5_28 
2022-12-07T15:23:05.8440260Z   libgfortran5       pkgs/main/osx-64::libgfortran5-11.3.0-h9dfd629_28 
2022-12-07T15:23:05.8440770Z   libnghttp2         pkgs/main/osx-64::libnghttp2-1.46.0-ha29bfda_0 
2022-12-07T15:23:05.8441560Z   libssh2            pkgs/main/osx-64::libssh2-1.10.0-h0a4fc7d_0 
2022-12-07T15:23:05.8442120Z   llvm-openmp        pkgs/main/osx-64::llvm-openmp-14.0.6-h0dcd299_0 
2022-12-07T15:23:05.8442590Z   ncurses            pkgs/main/osx-64::ncurses-6.3-hca72f7f_3 
2022-12-07T15:23:05.8443050Z   openssl            pkgs/main/osx-64::openssl-1.1.1s-hca72f7f_0 
2022-12-07T15:23:05.8443510Z   zlib               pkgs/main/osx-64::zlib-1.2.13-h4dc903c_0 
2022-12-07T15:23:05.8443680Z 
2022-12-07T15:23:05.8443690Z 
2022-12-07T15:23:05.8443810Z Preparing transaction: ...working... done
2022-12-07T15:23:05.8444070Z Verifying transaction: ...working... done
2022-12-07T15:23:05.8444550Z Executing transaction: ...working... Exception encountered: <pexpect.popen_spawn.PopenSpawn object at 0x7ff6cf543a90>
2022-12-07T15:23:05.8444880Z searcher: searcher_re:
2022-12-07T15:23:05.8445290Z     0: re.compile('Executing transaction: ...working... done.*\n')
2022-12-07T15:23:05.8445610Z <pexpect.popen_spawn.PopenSpawn object at 0x7ff6cf543a90>
2022-12-07T15:23:05.8445870Z searcher: searcher_re:
2022-12-07T15:23:05.8446240Z     0: re.compile('Executing transaction: ...working... done.*\n')
```
And both `test_powershell_basic_integration` and `test_zsh_basic_integration` failing for
```
h5stat --version
dyld[37821]: Library not loaded: '@rpath/libcrypto.1.1.dylib'
  Referenced from: '/Users/runner/conda_pkgs_dir/hdf5-1.12.1-h2b2ad87_2/lib/libhdf5.200.dylib'
  Reason: tried: '/Users/runner/conda_pkgs_dir/hdf5-1.12.1-h2b2ad87_2/lib/libcrypto.1.1.dylib' (no such file), '/Users/runner/conda_pkgs_dir/hdf5-1.12.1-h2b2ad87_2/bin/../lib/libcrypto.1.1.dylib' (no such file), '/Users/runner/conda_pkgs_dir/hdf5-1.12.1-h2b2ad87_2/bin/../lib/libcrypto.1.1.dylib' (no such file), '/usr/local/lib/libcrypto.1.1.dylib' (no such file), '/usr/lib/libcrypto.1.1.dylib' (no such file)
```
This could be related to some install/link issue. If `libcrypto.1.1.dylib` does not exist, that would mean `openssl` isn't linked properly -- however, notice the paths used for loading: those are all within the package cache and not `${CONDA_PREFIX}/bin/../lib/libcrypto.1.1.dylib`; so maybe rather a problem linking `hdf5`?

----
The second one is on Windows, failing with
```
tests/test_activate.py::test_stacking
```
```
PluginError: Error while loading conda plugins from entrypoints: DLL load failed: The specified module could not be found.
```
I don't see the exact cause for that error in the logs.
But it's another pitch for better error handling around plugins; ref: https://github.com/conda/conda/issues/12149 ;)

----
Is there some place where you collect failing/flaky tests for `conda`? Would be good to have one.
Would also be good to have the CI report (e.g., post comments to a special `CI test failures on <branch>` issue) test run failures of `main`/maintenance branches.

**Review by mbargull — APPROVED** [review]
LGTM!

**Pull request merged**